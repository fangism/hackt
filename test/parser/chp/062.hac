// "062.hac" should fail: no else clause allowed in do-while loop

defproc foo(chan?(bool) A, B, G; chan!(bool) Z) {
bool a, b, g;
chp {
	*[ a -> B?(b)
	[] ~a & b -> G?(g)
	[] else -> Z!(false)
	]
}
}

