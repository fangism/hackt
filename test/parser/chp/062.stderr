parse error: syntax error
parser stacks:
state	value
#STATE#	(null) 
#STATE#	(null) 
#STATE#	keyword: defproc [3:1..7]
#STATE#	identifier: foo [3:9..11]
#STATE#	list<(port-formal-decl)>: (port-formal-decl) ... [3:12..47]
#STATE#	{ [3:49]
#STATE#	list<(def-body-item)>: (instance-decl) ... [4:1..12]
#STATE#	keyword: chp [5:1..3]
#STATE#	{ [5:5]
#STATE#	*[ [6:2]
#STATE#	(chp-det-sel) [6:5--7:19]
#STATE#	[] [8:2]
#STATE#	keyword: else [8:5..8]
in state #STATE#, possible rules are:
	chp_unmatched_det_guarded_command_list : chp_unmatched_det_guarded_command_list THICKBAR . chp_guarded_command  (#RULE#)
acceptable tokens are: 
	'(' (shift)
	'-' (shift)
	'~' (shift)
	ID (shift)
	FLOAT (shift)
	INT (shift)
	SCOPE (shift)
	BOOL_TRUE (shift)
	BOOL_FALSE (shift)
